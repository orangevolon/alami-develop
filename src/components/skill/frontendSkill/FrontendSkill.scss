@import "../../../assets/styles/badge.scss";

.frontend-skill__article {
  .badges__div {
    height: 500px;
  }

  .html-badge__img {
    @include badge(1.4);

    left: 36%;
    top: 7%;
  }

  .css-badge__img {
    @include badge(1.6);

    left: 78%;
    top: 0;
  }

  .vue-badge__img {
    @include badge(1.3);

    left: 19%;
    top: 40%;
  }

  .redux-badge__img {
    @include badge(1.3);

    left: 38%;
    top: 63%;
  }

  .react-badge__img {
    @include badge(2.2);

    left: 82%;
    top: 44%;
  }

  .jest-badge__img {
    @include badge(1);

    left: 21%;
    top: 92%;
  }

  .js-badge__img {
    @include badge(1.3);

    left: 83%;
    top: 86%;
  }
}
