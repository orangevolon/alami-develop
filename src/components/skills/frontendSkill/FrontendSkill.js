import { Badge, Skill } from "@components";
import cssBadge from "@assets/badges/css.svg";
import htmlBadge from "@assets/badges/html.svg";
import vueBadge from "@assets/badges/vue.svg";
import reduxBadge from "@assets/badges/redux.svg";
import reactBadge from "@assets/badges/react.svg";
import jestBadge from "@assets/badges/jest.svg";
import jsBadge from "@assets/badges/js.svg";

import "./FrontendSkill.scss";

const FrontendSkill = () => {
  const cssBadgeElement = Badge({ src: cssBadge });
  cssBadgeElement.classList.add("css-badge__img");

  const htmlBadgeElement = Badge({ src: htmlBadge });
  htmlBadgeElement.classList.add("html-badge__img");

  const vueBadgeElement = Badge({ src: vueBadge });
  vueBadgeElement.classList.add("vue-badge__img");

  const reduxBadgeElement = Badge({ src: reduxBadge });
  reduxBadgeElement.classList.add("redux-badge__img");

  const reactBadgeElement = Badge({ src: reactBadge });
  reactBadgeElement.classList.add("react-badge__img");

  const jestBadgeElement = Badge({ src: jestBadge });
  jestBadgeElement.classList.add("jest-badge__img");

  const jsBadgeElement = Badge({ src: jsBadge });
  jsBadgeElement.classList.add("js-badge__img");

  const skillElement = Skill({
    badges: [
      cssBadgeElement,
      htmlBadgeElement,
      vueBadgeElement,
      reduxBadgeElement,
      reactBadgeElement,
      jestBadgeElement,
      jsBadgeElement
    ],
    title: "Front End Development",
    description: "Lorem Ipsum"
  });
  skillElement.classList.add("frontend-skill__article");

  return skillElement;
};

export default FrontendSkill;
