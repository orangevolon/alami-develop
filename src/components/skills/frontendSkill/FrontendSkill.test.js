import FrontendSkill from "./FrontendSkill";

describe("Testing components/skill/FrontendSkill", () => {
  it("Should render the component correctly", () => {
    expect(FrontendSkill()).toMatchSnapshot();
  });
});
