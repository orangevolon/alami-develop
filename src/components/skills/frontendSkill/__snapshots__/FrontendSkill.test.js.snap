// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components/skill/FrontendSkill Should render the component correctly 1`] = `
<article
  class="skill__article frontend-skill__article"
>
  <h3
    class="title__h3"
  />
  <p
    class="description__p"
  />
  <div
    class="badges__div"
  >
    <img
      class="badge__img css-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img html-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img vue-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img redux-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img react-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img jest-badge__img"
      src="[object Object]"
    />
    <img
      class="badge__img js-badge__img"
      src="[object Object]"
    />
  </div>
</article>
`;
